### PR TITLE
[Sprint 7 / PI - 109] Create Ninshiki Yolo Unit Tests

### PR DESCRIPTION
**Pull request by Nathan24zz**
## Jira Link: 
https://ichiro-its.atlassian.net/browse/PI-109
## Description

Create simple ninshiki_yolo unit tests, including detection tests and node initialization tests.

## Type of Change

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause the existing functionality to not work as expected)

## How Has This Been Tested?

- [ ] New unit tests added.
- [x] Manual tested.

## Checklist:

- [x] Using Branch Name Convention
    - `feature/JIRA-ID-SHORT-DESCRIPTION` if has a JIRA ticket
    - `enhancement/SHORT-DESCRIPTION` if has/has no JIRA ticket and contain enhancement
    - `hotfix/SHORT-DESCRIPTION` if the change doesn't need to be tested (urgent)
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made the documentation for the corresponding changes.

### CODE REVIEW

File: test/detector/test_detection.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    objects.append(Object('robot', 0.9772688746452332, 0.15625,
+                          0.13333333333333333, 0.2484375, 0.5666666666666667))
+    objects.append(Object('L-Intersection', 0.972011148929596, 0.275,
+                          0.21666666666666667, 0.325, 0.2722222222222222))
+    objects.append(Object('ball', 0.8741494417190552, 0.25625,
+                          0.4388888888888889, 0.3140625, 0.5388888888888889))
+    objects.append(Object('goalpost', 0.8019466400146484, 0.3828125,
+                          -0.011111111111111112, 0.425, 0.3333333333333333))
+    objects.append(Object('goalpost', 0.6941682696342468, 0.859375,
+                          0.044444444444444446, 0.996875, 0.7583333333333333))
```
Comment (segara2410) [inline test/detector/test_detection.py]: use round() with 2 decimals for these numbers

**Comment by segara2410** [comment]
add a GitHub workflow to check your test

File: .github/workflows/build-and-test-stable.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        with:
+          path: ninshiki_yolo
+
+      - name: Add stable Debian repository and rosdep sources list
+        run: |
+          sudo apt update && sudo apt install curl
+          curl -s http://repository.ichiro-its.org/debian/setup.bash | bash -s
+          curl -s http://repository.ichiro-its.org/rosdep/setup.bash | bash -s
+      - name: Build and test workspace
+        uses: ichiro-its/ros2-build-and-test-action@main
```
Comment (maroqijalil) [inline .github/workflows/build-and-test-stable.yml]: newline

File: .github/workflows/build-debian-nightly.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        uses: appleboy/ssh-action@master
+        with:
+          host: ${{ secrets.SSH_HOST }}
+          port: ${{ secrets.SSH_PORT }}
+          username: ${{ secrets.SSH_USER }}
+          password: ${{ secrets.SSH_PASS }}
+          script: |
+            cd ~/repository/debian
+            reprepro includedeb nightly ~/temp/nightly/ninshiki_yolo/package/*.deb
+            rm -rf ~/temp/nightly/ninshiki_yolo/
```
Comment (maroqijalil) [inline .github/workflows/build-debian-nightly.yml]: newline

File: .github/workflows/build-debian-stable.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        uses: appleboy/ssh-action@master
+        with:
+          host: ${{ secrets.SSH_HOST }}
+          port: ${{ secrets.SSH_PORT }}
+          username: ${{ secrets.SSH_USER }}
+          password: ${{ secrets.SSH_PASS }}
+          script: |
+            cd ~/repository/debian
+            reprepro includedeb stable ~/temp/stable/ninshiki_yolo/package/*.deb
+            rm -rf ~/temp/stable/ninshiki_yolo/
```
Comment (maroqijalil) [inline .github/workflows/build-debian-stable.yml]: newline

File: .github/workflows/build-and-test-nightly.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        with:
+          path: ninshiki_yolo
+
+      - name: Add nightly Debian repository and rosdep sources list
+        run: |
+          sudo apt update && sudo apt install curl
+          curl -s http://repository.ichiro-its.org/debian/setup-nightly.bash | bash -s
+          curl -s http://repository.ichiro-its.org/rosdep/setup.bash | bash -s
+      - name: Build and test workspace
+        uses: ichiro-its/ros2-build-and-test-action@main
```
Comment (maroqijalil) [inline .github/workflows/build-and-test-nightly.yml]: newline

File: package.xml [hunk 6]
```diff
@@ -9,8 +9,10 @@
   <exec_depend>ninshiki_interfaces</exec_depend>
   <exec_depend>shisen_interfaces</exec_depend>
   <exec_depend>rclpy</exec_depend>
-  <exec_depend>opencv-python</exec_depend>
-  <exec_depend>numpy</exec_depend>
+  <exec_depend>python-numpy</exec_depend>
+  <exec_depend>python-unittest2</exec_depend>
+  <exec_depend>python3-opencv</exec_depend>
+  <exec_depend>wget</exec_depend>
```
Comment (maroqijalil) [inline package.xml:15]: Maybe, you can order these dependencies (`exec_depend`) by alphabet with ascending order. (just suggestion)

**Review by maroqijalil — COMMENTED** [review]
Check my comments

File: .github/workflows/build-and-test-stable.yml (hunk 2 above)
Comment (Nathan24zz) [reply .github/workflows/build-and-test-stable.yml]: done

File: package.xml (hunk 6 above)
Comment (Nathan24zz) [reply package.xml:15]: done

**Review by segara2410 — APPROVED** [review]

**Review by maroqijalil — APPROVED** [review]
Lgtm

**Pull request merged**